packit_service/worker/handlers/copr.py

```diff
@@ -479,3 +496,156 @@ def handle_testing_farm(self):
+                "newly introduced defects in the `addded.html` in `Logs`. "
+                "You can disable the scanning in your configuration by "
+                "setting `osh_diff_scan_after_copr_build` to `False`. For more information, "
+                f"see [docs]({DOCS_URL}/configuration#osh_diff_scan_after_copr_build)."
+            ),
+        )
+
+        shutil.rmtree(directory)
+
+    def find_base_build_job(self) -> Optional[JobConfig]:
+        """
+        Find the job in the config that can provide the base build for the scan
+        (with `commit` trigger and same branch configured as the target PR branch).
+        """
+        base_build_job = None
+
+        for job in self.package_config.get_job_views():
+            if (
+                job.type in (JobType.copr_build, JobType.build)
+                and job.trigger == JobConfigTriggerType.commit
+                and (
+                    (
+                        job.branch
+                        and job.branch
+                        == self.copr_build_helper.pull_request_object.target_branch
+                    )
+                    or (
+                        not job.branch
+                        and self.project.default_branch
+                        == self.copr_build_helper.pull_request_object.target_branch
+                    )
+                )
+            ):
+                base_build_job = job
+                break
+
+        return base_build_job
+
+    def get_base_srpm_build(
+        self, base_build_job: JobConfig
+    ) -> Optional[SRPMBuildModel]:
+        """
+        Get the SRPM build model of the latest successful Copr build
+        for the given job config.
+        """
+        base_build_project_name = (
+            self.copr_build_helper.job_project_for_commit_job_config(base_build_job)
+        )
+        base_build_owner = self.copr_build_helper.job_owner_for_job_config(
+            base_build_job
+        )
+        target_branch_commit = (
+            self.copr_build_helper.pull_request_object.target_branch_head_commit
+        )
+
+        logger.debug(
+            f"Searching for base build for {target_branch_commit} commit "
+            f"in {base_build_owner}/{base_build_project_name} Copr project in our DB. "
+        )
+
+        builds = CoprBuildTargetModel.get_all_by(
+            commit_sha=target_branch_commit,
+            project_name=base_build_project_name,
+            owner=base_build_owner,
+            status=BuildStatus.success,
+        )
+        if not builds:
+            logger.debug("No matching base build found in our DB.")
+            return None
+
+        return list(builds)[0].get_srpm_build()
+
+    @staticmethod
+    def download_srpms(
+        directory: str,
+        base_srpm_build_model: SRPMBuildModel,
+        srpm_model: SRPMBuildModel,
+    ) -> Optional[tuple[Path, Path]]:
+        base_build_srpm_path = Path(directory).joinpath(
```

Comment:
Probably name the variable `base_srpm_path` to stay consistent with `srpm_path`.